pkg/controller/job/job_controller.go

```diff
@@ -643,7 +643,7 @@ func pastBackoffLimitOnFailure(job *batch.Job, pods []*v1.Pod) bool {
 			result += stat.RestartCount
 		}
 	}
-	return result >= *job.Spec.BackoffLimit
+	return result > *job.Spec.BackoffLimit
```

Comment:
Hmmm... I'm not sure this is correct. We are counting `RestartCount` explicitly. The pod that succeeds the first time does not have `RestartCount` greater than 0, in which case the case you're describing applies specifically to pods that did failed, either init containers, or regular containers in a pod.

Reply:
Right, but that only applies to the case when `BackoffLimit` is 0, so we need to change the condition accordingly. For `BackoffLimit` set to 1 and above, this won't be true, because only 1 backoff is allowed.